### PR TITLE
[2021.2] Fixed error with multisampled sampler in XR

### PR DESCRIPTION
**Pull request by thomas-zeng**
---
### Purpose of this PR
Fixed fogbugz case https://fogbugz.unity3d.com/f/cases/1297013/

---
### Testing status
- QA
- ABV

---
### Comments to reviewers
- URP non XR pass does not run copyDepth msaa code pass due to [CanCopyDepth ](https://github.com/Unity-Technologies/Graphics/blob/f5c82ce283b5b92ad0bebfcd0a0827c3b6404562/com.unity.render-pipelines.universal/Runtime/ForwardRenderer.cs#L821) check. In XR, we use copyDepth to support depth sharing in non-direct-to-eye case. Added check to configure samples to 1 when auto resolve is enabled or multisampled texture is not supported.

- To QA:
I have tested the repro project and verified the issue is fixed. The code impacts XR depth sharing area. Please consider to test other providers with depth sharing enabled and verify there is no regression in depth sharing area.

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/Passes/CopyDepthPass.cs [hunk 1]
```diff
@@ -63,7 +63,11 @@ public override void Execute(ScriptableRenderContext context, ref RenderingData
             using (new ProfilingScope(cmd, ProfilingSampler.Get(URPProfileId.CopyDepth)))
             {
                 RenderTextureDescriptor descriptor = renderingData.cameraData.cameraTargetDescriptor;
+
                 int cameraSamples = descriptor.msaaSamples;
+                // When auto resolve is supoorted or multisampled texture is not supported, set camera samples to 1
```
Comment (fabien-unity) [inline com.unity.render-pipelines.universal/Runtime/Passes/CopyDepthPass.cs:68]: small typo : supoorted

Comment (thomas-zeng) [reply com.unity.render-pipelines.universal/Runtime/Passes/CopyDepthPass.cs]: fixed in 0e6831a

**Review by fabien-unity — APPROVED** [review]

**Comment by ryanhy-unity** [comment]
Fix looks good.  Tested on my Quest and no black screen or errors.

**Review by ryanhy-unity — APPROVED** [review]

File: com.unity.render-pipelines.universal/Runtime/Passes/CopyDepthPass.cs [hunk 2]
```diff
@@ -63,7 +63,11 @@ public override void Execute(ScriptableRenderContext context, ref RenderingData
             using (new ProfilingScope(cmd, ProfilingSampler.Get(URPProfileId.CopyDepth)))
             {
                 RenderTextureDescriptor descriptor = renderingData.cameraData.cameraTargetDescriptor;
+
                 int cameraSamples = descriptor.msaaSamples;
+                // When auto resolve is supported or multisampled texture is not supported, set camera samples to 1
+                if (SystemInfo.supportsMultisampleAutoResolve || SystemInfo.supportsMultisampledTextures == 0)
+                    cameraSamples = 1;
```
Comment (ellioman) [inline com.unity.render-pipelines.universal/Runtime/Passes/CopyDepthPass.cs:70]: Can we maybe update the comment above for default case in the switch statement below?

**Review by ellioman — APPROVED** [review]

**Review by erikabar — CHANGES_REQUESTED** [review]
Android devices are still rendering black screen: 

tested and reproduced with:
- VLNQA00144, Samsung Galaxy S9 USA, 10.0.0, Snapdragon 845 SDM845, Adreno 630
- VLNQA00314, LGE Nexus 5X, 8.0.0, Snapdragon 808, Adreno 418
- VLNQA00320, Xiaomi Redmi Note 8 Pro, 9.0.0, MediaTek Helios G90T MT6785T, Mali-G76 MC4
- VLNQA00041, Samsung Galaxy S8, 8.0.0, Exynos 9 Octa 8895, Mali-G71
- VLNQA00175, Samsung Galaxy Note9, 10.0.0, Exynos 9 Series 9810, Mali-G72

**Review by erikabar — APPROVED** [review]
After discussing with Thomas, running mockHMD on Android shows black screen is actually expected. I am not getting errors as described in the bug report.

**Review by phi-lira — APPROVED** [review]

**Pull request merged**